From d765d82c1d661b038cfc6dc87f5a3b3a49181200 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tryggvi=20Bj=C3=B6rgvinsson?= <tryggvib@fsfi.is>
Date: Thu, 20 Sep 2012 09:38:26 +0000
Subject: [PATCH] Added support to load Jinja2 templates from a list of paths
 (as supported by FileSystemLoader). When initiating the
 template engine, pass in a list of paths (strings) instead
 of a single string.

---
 txtemplate/templates.py |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/txtemplate/templates.py b/txtemplate/templates.py
index cc1558d..04a03c5 100644
--- a/txtemplate/templates.py
+++ b/txtemplate/templates.py
@@ -253,14 +253,18 @@ class Jinja2TemplateLoader(object):
     interface.implements(itemplate.ITemplateLoader)
 
     def __init__(self, path, **options):
-        self.path = path
+        if not path:
+            self.path = os.curdir
+        elif isinstance(path, list):
+            self.path = [os.path.join(os.path.abspath(p)) for p in path]
+        else:
+            self.path = os.path.join(os.path.abspath(path))
+
         self.options = {"auto_reload": False}
         self.options.update(options)
-        if not self.path:
-            self.path = os.curdir
 
         self.loader = jinja2.FileSystemLoader(
-            os.path.join(os.path.abspath(self.path)),
+            self.path,
             encoding="utf-8"
         )
         self.environment = jinja2.Environment(loader=self.loader, **self.options)
-- 
1.7.9.5

